.github/actions/start-opensearch/action.yml

```diff
@@ -34,7 +35,15 @@ runs:
 
         if [[ -d "$OPENSEARCH_HOME/plugins/opensearch-security" ]]; then
           if [[ "$SECURED" == "true" ]]; then
-            bash $OPENSEARCH_HOME/plugins/opensearch-security/tools/install_demo_configuration.sh -y -i -s
+            OPENSEARCH_VERSION=$(cat $OPENSEARCH_HOME/plugins/opensearch-security/plugin-descriptor.properties | grep '^version=' | cut -d'=' -f 2)
+            OPENSEARCH_REQUIRED_VERSION="2.12.0"
+            # Starting in 2.12.0, security demo configuration script requires an initial admin password
+            COMPARE_VERSION=`echo $OPENSEARCH_REQUIRED_VERSION $OPENSEARCH_VERSION | tr ' ' '\n' | sort -V | uniq | head -n 1`
+            if [ "$COMPARE_VERSION" == "$OPENSEARCH_REQUIRED_VERSION" ]; then
+              bash $OPENSEARCH_HOME/plugins/opensearch-security/tools/install_demo_configuration.sh -y -i -s -t
```

Comment:
Ideally >=2.12.0 should use a strong password, but since this repo is using the scriptfile, it is possible to add this `-t` parameter and still use admin. This minimizes the surface area of this change.